packages/kbn-config/src/config_service.ts

```diff
@@ -111,13 +119,16 @@ export class ConfigService {
    *
    * This must be done after every schemas and deprecation providers have been registered.
    */
-  public async validate() {
+  public async validate(params: ConfigValidateParameters = { logDeprecations: true }) {
```

Comment:
__note:__ I'm not really happy about this change, but couldn't find a better option - we need to validate entire config twice: during `preboot` stage and during `setup` stage as the config can change between these stages. But we don't want to record deprecation warnings twice and it also broke a bunch of "usage" functional tests because of these duplicated information.

Reply:
> We can't avoid config validation though, meaning that the initial validation will throw an error in case of invalid configuration, but I think that's what we want anyway?

Yep, that's intentional, we should shut Kibana down as soon as we can when we detect invalid configuration

Reply:
The idea is that `preboot` plugins may need to change config, and the only reliable way for the Core to know that config is still valid in `setup` is to re-validate everything again. For example, in interactive setup mode user will need to configure connection to ES and we'll have to write this configuration this to the disk, we may also allow them to configure things provided by other plugins (e.g. encryption keys for Security/Reporting/ESO).